tests/displays/cavity_display/backend/test_backend_cavity.py

```diff
@@ -146,33 +150,105 @@
 
 
 def mock_open(*args, **kwargs):
-    data = [",".join(csv_keys), ",".join(csv_rack_row)]
+    data = [
+        ",".join(csv_keys),
+        ",".join(csv_rack_row),
+        ",".join(csv_all_row),
+        ",".join(csv_ssa_row),
+        ",".join(csv_cryo_row),
+        ",".join(csv_cm_row),
+    ]
     # mocked open for path "foo"
     return mock.mock_open(read_data="\n".join(data))(*args, **kwargs)
 
 
-# These import states try to read from faults.csv which causes an error
-with mock.patch("builtins.open", mock_open):
-    from displays.cavity_display.backend.backend_cavity import BackendCavity
-
-    DISPLAY_MACHINE = Machine(
-        cavity_class=BackendCavity,
-    )
-
-
 class TestBackendCavity(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.mock_open_patcher = mock.patch("builtins.open", mock_open) # We are setting up the mock for the test class. (Replacing the open func)
+        cls.mock_open_patcher.start()
+        # Create display Machine
+        cls.DISPLAY_MACHINE = Machine(cavity_class=BackendCavity)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.mock_open_patcher.stop() # Stops patcher when we finish all tests
+
+    def setUp(self):
+        self.cm01 = self.DISPLAY_MACHINE.cryomodules["01"]
```

Comment:
Hmm I feel like this doesn't need to be in the setup. We can probably use a cavity iterator in each function... Lemme think